x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsCanMatchOnCoordinatorIntegTests.java

```diff
@@ -0,0 +1,288 @@
+                        .to("2020-11-29T00:00:00.000000000Z")
+                )
+            );
+
+        if (includeIndexCoveringSearchRangeInSearchRequest) {
+            SearchResponse searchResponse = client().search(request).actionGet();
+
+            // All the regular index searches succeeded
+            assertThat(searchResponse.getSuccessfulShards(), equalTo(indexWithinSearchRangeShardCount));
+            // All the searchable snapshots shard search failed
+            assertThat(searchResponse.getFailedShards(), equalTo(indexOutsideSearchRangeShardCount));
+            assertThat(searchResponse.getSkippedShards(), equalTo(0));
+            assertThat(searchResponse.getTotalShards(), equalTo(totalShards));
+        } else {
+            // All shards failed, since all shards are unassigned and the IndexMetadata min/max timestamp
+            // is not available yet
+            expectThrows(SearchPhaseExecutionException.class, () -> client().search(request).actionGet());
+        }
+
+        // Allow the searchable snapshots to be finally mounted
+        unblockNode(repositoryName, dataNodeHoldingSearchableSnapshot);
+        waitUntilRecoveryIsDone(searchableSnapshotIndexOutsideSearchRange);
+        ensureGreen(searchableSnapshotIndexOutsideSearchRange);
+
+        final IndexMetadata updatedIndexMetadata = getIndexMetadata(searchableSnapshotIndexOutsideSearchRange);
+        final IndexLongFieldRange updatedTimestampMillisRange = updatedIndexMetadata.getTimestampMillisRange();
+        assertThat(updatedTimestampMillisRange.isComplete(), equalTo(true));
+        assertThat(updatedTimestampMillisRange, not(sameInstance(IndexLongFieldRange.EMPTY)));
+        assertThat(updatedTimestampMillisRange.getMin(), greaterThanOrEqualTo(Instant.parse("2020-11-26T00:00:00Z").getMillis()));
+        assertThat(updatedTimestampMillisRange.getMax(), lessThanOrEqualTo(Instant.parse("2020-11-27T00:00:00Z").getMillis()));
+        assertThat(indicesService.getTimestampFieldType(updatedIndexMetadata.getIndex()), notNullValue());
+
+        // Stop the node holding the searchable snapshots, and since we defined
+        // the index allocation criteria to require the searchable snapshot
+        // index to be allocated in that node, the shards should remain unassigned
+        internalCluster().stopNode(dataNodeHoldingSearchableSnapshot);
+        waitUntilAllShardsAreUnassigned(updatedIndexMetadata.getIndex());
+
+        if (includeIndexCoveringSearchRangeInSearchRequest) {
+            SearchResponse newSearchResponse = client().search(request).actionGet();
+
+            assertThat(newSearchResponse.getSkippedShards(), equalTo(indexOutsideSearchRangeShardCount));
+            assertThat(newSearchResponse.getSuccessfulShards(), equalTo(totalShards));
+            assertThat(newSearchResponse.getFailedShards(), equalTo(0));
+            assertThat(newSearchResponse.getTotalShards(), equalTo(totalShards));
```

Comment:
Perhaps also assert that we get the right number of total hits?

Reply:
Fixed in https://github.com/elastic/elasticsearch/pull/65583/commits/eada99ea3b08b21ef9ce5a39e93a305331bf9117